runtimevar/example_test.go

```diff
@@ -72,3 +72,39 @@ func Example_stringVariable() {
 	// Output:
 	// "hello world"
 }
+
+func ExampleVariable_Watch() {
+	// Construct a *Variable using a constructor from one of the
+	// runtimevar subpackages. This example uses constantvar.
+	// The variable value is of type string, so we use StringDecoder.
+	v := constantvar.NewBytes([]byte("hello world"), runtimevar.StringDecoder)
+	defer v.Close()
+
+	// Call Watch in a loop from a background goroutine to see all changes,
+	// including errors.
+	ctx := context.Background()
+	ch := make(chan struct{})
+	go func() {
+		for {
+			if snapshot, err := v.Watch(ctx); err == nil {
+				fmt.Printf("New config: %v", snapshot.Value.(string))
+				close(ch)
+			} else if err != runtimevar.ErrClosed {
+				fmt.Printf("Error loading config: %v", err)
+				// Even though there's been an error loading the config,
+				// v.Latest will continue to return the latest "good" value.
+			} else {
+				// v has been closed; exit.
+				return
```

Comment:
And `ch` is not closed? So what happens in the containing function?

Reply:
The flow will be that first `Watch` will return, closing `ch`; that allows the containing function to get past the `<-ch` and exit, closing v.

This may be too confusing for an example. Maybe I should remove the `// Output` lines and just have it compile?

Reply:
Yeah, I'm still a bit confused. What if the first call to `Watch` returns `runtimevar.ErrClosed`? There is a nontrivial protocol with the wrapping function here, could it be simplified? Or at least documented with comments

Reply:
OK. I removed the channel and update the test to not run, hopefully it make more sense now.